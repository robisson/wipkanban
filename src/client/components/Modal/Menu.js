import React from 'react';

const Menu = () => {
    return (
        <div>
            <a id="addChecklist" href="#" className="btn btn-sm btn-primary">
                <span className="fa fa-check-square"></span>
                Checklist
            </a>
            <span id="containerDataTarefa">
                <span>
                    <a href="#" id="btnMostrarDatepicker" className="hide btn btn-sm btn-primary">
                        <span className="fa fa-calendar"></span>
                        Definir data
                    </a>
                    <span id="datePickerDataTarefa" className="hide">
                        <input
                            size="8"
                            maxLength="10"
                            type="text"
                            defaultValue="14/08/2017"
                            className="datepicker text-center"/>
                        <button id="btnDefinirData" className="btn btn-success btn-sm">OK</button>
                    </span>
                    <span className="" id="dataSelecionada">
                        <span className="badge badge-success">
                            <input type="checkbox"/>
                            <span title="Clique para alterar a data" id="dataTarefa">14/08/2017</span>
                        </span>
                        <button id="btnRemover" className="btn btn-sm btn-danger" title="Remover data">
                            <i className="fa fa-remove"></i>
                        </button>
                    </span>
                </span>
            </span>
            <a id="btnMarcarTempo" href="#" className="btn btn-sm btn-primary">
                <span className="fa fa-clock-o"></span>
                Marcar tempo
            </a>
            <a href="#" className="btn btn-sm btn-danger" id="excluirTarefa">
                <span className="fa fa-remove"></span>
                Excluir
            </a>
        </div>
    );
}

export default Menu;