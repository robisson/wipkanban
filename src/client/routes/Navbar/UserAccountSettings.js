import * as React from "./react";
import Tooltip from '@material-ui/core/Tooltip';
import IconButton from '@material-ui/core/IconButton';
import Avatar from '@material-ui/core/Avatar';

const UserAccountSettings = () => {
  return (
    <Tooltip title="My account settings">
      <IconButton color="inherit" aria-label="My account settings">
        <Avatar
          aria-label="Robisson Oliveira"
          src="/uploads/7b435896-aeca-4660-aaa3-b734e3e9200a.jpg"
        />
      </IconButton>
    </Tooltip>
  );
};

export default UserAccountSettings;
