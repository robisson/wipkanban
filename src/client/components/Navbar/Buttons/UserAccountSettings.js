import * as React from "react";
import { withStyles } from "@material-ui/core/styles";
import IconButton from "@material-ui/core/IconButton";
import Avatar from "@material-ui/core/Avatar";
import Menu from "@material-ui/core/Menu";
import MenuItem from "@material-ui/core/MenuItem";
import ListItemIcon from "@material-ui/core/ListItemIcon";
import CallSplitIcon from "@material-ui/icons/CallSplit";
import ListItemText from "@material-ui/core/ListItemText";

const ITEM_HEIGHT = 48;

const styles: Object = {
  menuItem: {
    paddingTop: 2,
    paddingBottom: 2
  }
};

type State = {
  anchorEl?: Object
};

type Props = {
  classes: Object,
  onLogout: Function,
  user: Object
};

class UserAccountSettings extends React.Component<Props, State> {
  state: State = {
    anchorEl: null
  };

  handleClick = event => {
    this.setState({ anchorEl: event.currentTarget });
  };

  handleClose = () => {
    this.setState({ anchorEl: null });
  };

  render() {
    const { anchorEl } = this.state;
    const { classes, onLogout, user } = this.props;

    return (
      <span>
        <IconButton
          onClick={this.handleClick}
          color="inherit"
          aria-label="My account settings"
        >
          <Avatar
            aria-label={`${user.name} ${user.lastname}`}
            src={user.image}
          />
        </IconButton>
        <Menu
          id="long-menu"
          anchorEl={anchorEl}
          open={Boolean(anchorEl)}
          onClose={this.handleClose}
          PaperProps={{
            style: {
              maxHeight: ITEM_HEIGHT * 4.5,
              width: 200
            }
          }}
        >
          <MenuItem onClick={this.handleClose} className={classes.menuItem}>
            <ListItemIcon>
              <CallSplitIcon />
            </ListItemIcon>
            <ListItemText inset primary="Vertical Split" />
          </MenuItem>
          <MenuItem onClick={this.handleClose} className={classes.menuItem}>
            <ListItemIcon>
              <CallSplitIcon />
            </ListItemIcon>
            <ListItemText inset primary="Sair" onClick={() => onLogout()} />
          </MenuItem>
        </Menu>
      </span>
    );
  }
}

export default withStyles(styles)(UserAccountSettings);
