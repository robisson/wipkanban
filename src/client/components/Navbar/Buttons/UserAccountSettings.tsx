import * as React from "react";
import { withStyles, StyleRulesCallback } from "@material-ui/core/styles";
import IconButton from "@material-ui/core/IconButton";
import Avatar from "@material-ui/core/Avatar";
import Menu from "@material-ui/core/Menu";
import MenuItem from "@material-ui/core/MenuItem";
import ListItemIcon from "@material-ui/core/ListItemIcon";
import CallSplitIcon from "@material-ui/icons/CallSplit";
import ListItemText from "@material-ui/core/ListItemText";

const ITEM_HEIGHT = 48;

const styles: StyleRulesCallback = () => {
  return {
    menuItem: {
      paddingTop: 2,
      paddingBottom: 2
    }
  };
};

interface IUserAccountSettingsState {
  anchorEl?: any;
}

interface IUserAccountSettingsProps {
  classes: any;
  onLogout: Function;
  user: any;
}

class UserAccountSettings extends React.Component<
  IUserAccountSettingsProps,
  IUserAccountSettingsState
> {
  state: IUserAccountSettingsState = {
    anchorEl: null
  };

  handleClick = (event: any) => {
    this.setState({ anchorEl: event.currentTarget });
  };

  handleClose = () => {
    this.setState({ anchorEl: null });
  };

  render() {
    const { anchorEl } = this.state;
    const { classes, onLogout, user } = this.props;

    return (
      <span>
        <IconButton
          onClick={this.handleClick}
          color="inherit"
          aria-label="My account settings"
        >
          <Avatar
            aria-label={`${user.name} ${user.lastname}`}
            src={user.image}
          />
        </IconButton>
        <Menu
          id="long-menu"
          anchorEl={anchorEl}
          open={Boolean(anchorEl)}
          onClose={this.handleClose}
          PaperProps={{
            style: {
              maxHeight: ITEM_HEIGHT * 4.5,
              width: 200
            }
          }}
        >
          <MenuItem onClick={this.handleClose} className={classes.menuItem}>
            <ListItemIcon>
              <CallSplitIcon />
            </ListItemIcon>
            <ListItemText inset primary="Vertical Split" />
          </MenuItem>
          <MenuItem onClick={this.handleClose} className={classes.menuItem}>
            <ListItemIcon>
              <CallSplitIcon />
            </ListItemIcon>
            <ListItemText inset primary="Sair" onClick={() => onLogout()} />
          </MenuItem>
        </Menu>
      </span>
    );
  }
}

export default withStyles(styles)(UserAccountSettings);
