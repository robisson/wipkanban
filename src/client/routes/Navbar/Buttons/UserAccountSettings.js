import * as React from "react";
import { withStyles } from "@material-ui/core/styles";
import IconButton from "@material-ui/core/IconButton";
import Avatar from "@material-ui/core/Avatar";
import Menu from "@material-ui/core/Menu";
import MenuItem from "@material-ui/core/MenuItem";
import ListItemIcon from "@material-ui/core/ListItemIcon";
import CallSplitIcon from "@material-ui/icons/CallSplit";
import ListItemText from "@material-ui/core/ListItemText";
import { logout } from "../../../actions/Login";
import { connect } from "react-redux";

const ITEM_HEIGHT = 48;

const styles: Object = {
  menuItem: {
    paddingTop: 2,
    paddingBottom: 2
  }
};

type State = {
  anchorEl?: Object
};

type Props = {
  classes: Object,
  logout:Function
};

class UserAccountSettings extends React.Component<Props, State> {
  state: State = {
    anchorEl: null
  };

  handleClick = event => {
    this.setState({ anchorEl: event.currentTarget });
  };

  handleClose = () => {
    this.setState({ anchorEl: null });
  };

  render() {
    const { anchorEl } = this.state;
    const { classes,logout } = this.props;
    return (
      <span>
        <IconButton
          onClick={this.handleClick}
          color="inherit"
          aria-label="My account settings"
        >
          <Avatar
            aria-label="Robisson Oliveira"
            src="/uploads/7b435896-aeca-4660-aaa3-b734e3e9200a.jpg"
          />
        </IconButton>
        <Menu
          id="long-menu"
          anchorEl={anchorEl}
          open={Boolean(anchorEl)}
          onClose={this.handleClose}
          PaperProps={{
            style: {
              maxHeight: ITEM_HEIGHT * 4.5,
              width: 200
            }
          }}
        >
          <MenuItem onClick={this.handleClose} className={classes.menuItem}>
            <ListItemIcon>
              <CallSplitIcon />
            </ListItemIcon>
            <ListItemText inset primary="Vertical Split" />
          </MenuItem>
          <MenuItem onClick={this.handleClose} className={classes.menuItem}>
            <ListItemIcon>
              <CallSplitIcon />
            </ListItemIcon>
            <ListItemText inset primary="Sair" onClick={logout} />
          </MenuItem>
        </Menu>
      </span>
    );
  }
}

export default withStyles(styles)(
  connect(undefined, { logout })(UserAccountSettings)
);
