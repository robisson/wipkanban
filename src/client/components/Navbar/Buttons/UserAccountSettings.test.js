import * as React from "react";
import UserAccountSettings from "./UserAccountSettings";
import { createShallow } from "@material-ui/core/test-utils";
import toJson from "enzyme-to-json";
import "../../../setupTest";
import Menu from "@material-ui/core/Menu";
import MenuItem from "@material-ui/core/MenuItem";
import ListItemText from "@material-ui/core/ListItemText";
import IconButton from "@material-ui/core/IconButton";

describe("<UserAccountSettings /> Component", () => {
  let shallow;
  let wrapper;
  let user;
  let fn;
  beforeEach(() => {
    shallow = createShallow({ dive: true });
    user = { name: "name", lastname: "lastname" };
    fn = jest.fn();
    wrapper = shallow(<UserAccountSettings user={user} onLogout={fn} />);
  });

  it("should render correctly", () => {
    expect(wrapper.find(Menu).type()).toEqual(Menu);
    expect(wrapper.find(MenuItem).length).toEqual(2);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should state is not null when clicked on IconBUtton button", () => {
    wrapper.find(IconButton).simulate("click", {
      currentTarget: true
    });

    expect(wrapper.state()).toEqual({anchorEl: true});

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should state to be null when clicked on MenuItem", () => {
    wrapper.find(MenuItem).at(0).simulate("click");

    expect(wrapper.state()).toBeNull;

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should call function when clicked on logout button", () => {
    wrapper
      .find(ListItemText)
      .at(1)
      .simulate("click");
    expect(fn.mock.calls.length).toEqual(1);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
