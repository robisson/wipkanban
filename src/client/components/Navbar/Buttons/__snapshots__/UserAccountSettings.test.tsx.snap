// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserAccountSettings /> Component should call function when clicked on logout button 1`] = `
<span>
  <WithStyles(IconButton)
    aria-label="My account settings"
    color="inherit"
    onClick={[Function]}
  >
    <WithStyles(Avatar)
      aria-label="name lastname"
    />
  </WithStyles(IconButton)>
  <WithStyles(Menu)
    PaperProps={
      Object {
        "style": Object {
          "maxHeight": 216,
          "width": 200,
        },
      }
    }
    anchorEl={null}
    id="long-menu"
    onClose={[Function]}
    open={false}
  >
    <WithStyles(MenuItem)
      className="UserAccountSettings-menuItem-1"
      onClick={[Function]}
    >
      <WithStyles(ListItemIcon)>
        <pure(CallSplitIcon) />
      </WithStyles(ListItemIcon)>
      <WithStyles(ListItemText)
        inset={true}
        primary="Vertical Split"
      />
    </WithStyles(MenuItem)>
    <WithStyles(MenuItem)
      className="UserAccountSettings-menuItem-1"
      onClick={[Function]}
    >
      <WithStyles(ListItemIcon)>
        <pure(CallSplitIcon) />
      </WithStyles(ListItemIcon)>
      <WithStyles(ListItemText)
        inset={true}
        onClick={[Function]}
        primary="Sair"
      />
    </WithStyles(MenuItem)>
  </WithStyles(Menu)>
</span>
`;

exports[`<UserAccountSettings /> Component should render correctly 1`] = `
<span>
  <WithStyles(IconButton)
    aria-label="My account settings"
    color="inherit"
    onClick={[Function]}
  >
    <WithStyles(Avatar)
      aria-label="name lastname"
    />
  </WithStyles(IconButton)>
  <WithStyles(Menu)
    PaperProps={
      Object {
        "style": Object {
          "maxHeight": 216,
          "width": 200,
        },
      }
    }
    anchorEl={null}
    id="long-menu"
    onClose={[Function]}
    open={false}
  >
    <WithStyles(MenuItem)
      className="UserAccountSettings-menuItem-1"
      onClick={[Function]}
    >
      <WithStyles(ListItemIcon)>
        <pure(CallSplitIcon) />
      </WithStyles(ListItemIcon)>
      <WithStyles(ListItemText)
        inset={true}
        primary="Vertical Split"
      />
    </WithStyles(MenuItem)>
    <WithStyles(MenuItem)
      className="UserAccountSettings-menuItem-1"
      onClick={[Function]}
    >
      <WithStyles(ListItemIcon)>
        <pure(CallSplitIcon) />
      </WithStyles(ListItemIcon)>
      <WithStyles(ListItemText)
        inset={true}
        onClick={[Function]}
        primary="Sair"
      />
    </WithStyles(MenuItem)>
  </WithStyles(Menu)>
</span>
`;

exports[`<UserAccountSettings /> Component should state is not null when clicked on IconBUtton button 1`] = `
<span>
  <WithStyles(IconButton)
    aria-label="My account settings"
    color="inherit"
    onClick={[Function]}
  >
    <WithStyles(Avatar)
      aria-label="name lastname"
    />
  </WithStyles(IconButton)>
  <WithStyles(Menu)
    PaperProps={
      Object {
        "style": Object {
          "maxHeight": 216,
          "width": 200,
        },
      }
    }
    anchorEl={true}
    id="long-menu"
    onClose={[Function]}
    open={true}
  >
    <WithStyles(MenuItem)
      className="UserAccountSettings-menuItem-1"
      onClick={[Function]}
    >
      <WithStyles(ListItemIcon)>
        <pure(CallSplitIcon) />
      </WithStyles(ListItemIcon)>
      <WithStyles(ListItemText)
        inset={true}
        primary="Vertical Split"
      />
    </WithStyles(MenuItem)>
    <WithStyles(MenuItem)
      className="UserAccountSettings-menuItem-1"
      onClick={[Function]}
    >
      <WithStyles(ListItemIcon)>
        <pure(CallSplitIcon) />
      </WithStyles(ListItemIcon)>
      <WithStyles(ListItemText)
        inset={true}
        onClick={[Function]}
        primary="Sair"
      />
    </WithStyles(MenuItem)>
  </WithStyles(Menu)>
</span>
`;

exports[`<UserAccountSettings /> Component should state to be null when clicked on MenuItem 1`] = `
<span>
  <WithStyles(IconButton)
    aria-label="My account settings"
    color="inherit"
    onClick={[Function]}
  >
    <WithStyles(Avatar)
      aria-label="name lastname"
    />
  </WithStyles(IconButton)>
  <WithStyles(Menu)
    PaperProps={
      Object {
        "style": Object {
          "maxHeight": 216,
          "width": 200,
        },
      }
    }
    anchorEl={null}
    id="long-menu"
    onClose={[Function]}
    open={false}
  >
    <WithStyles(MenuItem)
      className="UserAccountSettings-menuItem-1"
      onClick={[Function]}
    >
      <WithStyles(ListItemIcon)>
        <pure(CallSplitIcon) />
      </WithStyles(ListItemIcon)>
      <WithStyles(ListItemText)
        inset={true}
        primary="Vertical Split"
      />
    </WithStyles(MenuItem)>
    <WithStyles(MenuItem)
      className="UserAccountSettings-menuItem-1"
      onClick={[Function]}
    >
      <WithStyles(ListItemIcon)>
        <pure(CallSplitIcon) />
      </WithStyles(ListItemIcon)>
      <WithStyles(ListItemText)
        inset={true}
        onClick={[Function]}
        primary="Sair"
      />
    </WithStyles(MenuItem)>
  </WithStyles(Menu)>
</span>
`;
